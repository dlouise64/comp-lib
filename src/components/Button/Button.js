import React from 'react';
import PropTypes from 'prop-types';

import './Button.css';

class Button extends React.Component {
  render() {
    const {message} = this.props;
    return (
      <input className="btn info" type="submit" value={message} />
    );
  }
}

Button.propTypes = {
  /** message to display on button */
  message: PropTypes.string.isRequired
};

Button.defaultProps = {
  /** Default message */
  message: "Submit"
}

export default Button;
