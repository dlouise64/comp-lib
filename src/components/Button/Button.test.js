import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

test('<Button /> renders correctly', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
