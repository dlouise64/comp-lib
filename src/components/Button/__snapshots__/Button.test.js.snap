// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<input
  className="btn info"
  type="submit"
  value="Submit"
/>
`;
